### PR TITLE
Fade-in/Fade-out

### PR DESCRIPTION
**Pull request by ColdVolcano**
Temporary solution and no more than that. Not sure if behavior is acceptable

### CODE REVIEW

File: osu.Framework/Audio/Track/AudioTrackBass.cs [hunk 1]
```diff
@@ -76,6 +76,17 @@ public override void Reset()
             base.Reset();
         }
 
+        public override void FadeIn(double start = 0, int time = 100)
```
Comment (peppy) [inline osu.Framework/Audio/Track/AudioTrackBass.cs]: why is start a thing?

File: osu.Framework/Audio/Track/AudioTrackVirtual.cs [hunk 2]
```diff
@@ -42,6 +43,16 @@ public override void Stop()
             clock.Stop();
         }
 
+        public override void FadeIn(double start, int time)
+        {
+            throw new NotImplementedException();
```
Comment (peppy) [inline osu.Framework/Audio/Track/AudioTrackVirtual.cs]: just noop instead of exception?

File: osu.Framework/Audio/Track/AudioTrackBass.cs (hunk 1 above)
Comment (peppy) [inline osu.Framework/Audio/Track/AudioTrackBass.cs]: time should be double, and should probably be called length

Comment (ColdVolcano) [inline osu.Framework/Audio/Track/AudioTrackBass.cs]: 4th argument of `Bass.ChannelSlideAttribute` is Int32, could cast if said.
As for start... it's the starting volume (which is not nessesary I guess)

File: osu.Framework/Audio/Track/AudioTrackVirtual.cs (hunk 2 above)
Comment (ColdVolcano) [inline osu.Framework/Audio/Track/AudioTrackVirtual.cs]: 👍

File: osu.Framework/Audio/Track/AudioTrack.cs [hunk 3]
```diff
@@ -59,6 +59,21 @@ public virtual void Reset()
 
         public abstract void Stop();
 
+        /// <summary>
+        /// Fade-in the actual playing track.
+        /// </summary>
+        /// <param name="start">From where to start Fading</param>
+        /// <param name="lenght">The lenght in miliseconds that should take the Fade</param>
+        /// <returns>Whether the fade was succesful</returns>
+        public abstract void FadeIn(double lenght, double start);
```
Comment (peppy) [inline osu.Framework/Audio/Track/AudioTrack.cs]: length not lenght

also you haven't removed start

File: osu.Framework/Audio/Track/AudioTrackBass.cs [hunk 4]
```diff
@@ -76,6 +76,17 @@ public override void Reset()
             base.Reset();
         }
 
+        public override void FadeIn(double length = 100)
```
Comment (peppy) [inline osu.Framework/Audio/Track/AudioTrackBass.cs]: no default

File: osu.Framework/Audio/Track/AudioTrackBass.cs [hunk 5]
```diff
@@ -76,6 +76,17 @@ public override void Reset()
             base.Reset();
         }
 
+        public override void FadeIn(double length = 100)
+        {
+            Bass.ChannelSetAttribute(activeStream, ChannelAttribute.Volume, 0);
```
Comment (peppy) [inline osu.Framework/Audio/Track/AudioTrackBass.cs]: no

**Comment by peppy** [comment]
Closing for time being as invalid 👾

**Pull request closed without merging**